php/libraries/LorisForm.class.inc

```diff
@@ -1180,7 +1180,7 @@ class LorisForm
 
                 if (isset($el['required'])
                     && $el['required'] === true
-                    && empty($value)
+                    && (is_null($value) || (empty($value) && $value !=0))
```

Comment:
Could this just read `&& $value)`? This will evaluate to `false` in each of the cases you want here. 

http://stackoverflow.com/questions/2382490/how-does-true-false-work-in-php#2382510